### ISSUE TITLE
lintr:::get_source_expressions fails when a code comment contains an r script command surrounded by backticks

### ISSUE THREAD

**Issue by kpagacz** [r-lib/lintr#879]
We were surprised our files linted correctly despite having obvious linting issues, so I have investigated and narrowed it down to this minimal reprex:
```r
# `r print("7")`
function() 2<=3
```
The above doesn't lint.

I further narrowed the issue down to backticks in combination with r and something because the following do lint correctly:
```r
# `print("7")`
function() 2<=3

# `r`
function() 2<=3

# r print("7")
function() 2<=3
```
In short - all three are needed: backticks, r and something in addition to r.

Then, I dug inside inflix_spaces_linter and tracked the issue to the parsing function, Given a file "test.R", `lintr:::get_source_expressions("test.R")` returns:
![image](https://user-images.githubusercontent.com/24480246/138069946-9daa9966-cd15-4f04-a30d-78441dd8272c.png)
Without the comment, it parses correctly:
![image](https://user-images.githubusercontent.com/24480246/138070384-18bab982-8dad-46e8-9f19-7683f74f9686.png)

Environment:
lintr version 2.0.1
R: 4.1.0

Is this a known issue?

**Comment by MichaelChirico** [comment]
I believe it's a duplicate of #743
